src/olympia/versions/models.py

```diff
@@ -142,8 +142,8 @@ def from_upload(cls, upload, addon, platforms, channel, send_signal=True,
             parsed_data = utils.parse_addon(upload, addon)
         license_id = None
         if channel == amo.RELEASE_CHANNEL_LISTED:
-            previous_version = addon.find_latest_version_including_rejected(
-                channel=channel)
+            previous_version = addon.find_latest_version(
+                channel=channel, exclude=())
```

Comment:
There's an argument to exclude Beta because after #4529 beta versions don't get the details step where we'd ask for confirmation of the license.  (Though they'd get one anyway because of lines below).